jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/RequestLogger.java

```diff
@@ -141,22 +157,57 @@ public static void logAccess(HttpServerRequest request, HttpServerResponse respo
      * @param requestStartTime Duration of the request
      * @param log              The logger to write to
      */
-    public static void logRequestResponse(HttpServerRequest request, HttpServerResponse response, long requestStartTime, Logger log) {
-        long          duration = System.currentTimeMillis() - requestStartTime;
-        StringBuilder logLine  = new StringBuilder();
-        RequestLogger.logAccess(request, response, duration, logLine);
+    public void logRequestResponse(HttpServerRequest request, HttpServerResponse response, long requestStartTime, Logger log) {
+        long duration = System.currentTimeMillis() - requestStartTime;
+        StringBuilder logLine = new StringBuilder();
+        logAccess(request, response, duration, logLine);
         if (log.isDebugEnabled()) {
             logLine.append(" Headers: ");
-            RequestLogger.headersToString(request, logLine);
+            headersToString(request, logLine);
             logLine.append(" Response Headers: ");
-            RequestLogger.headersToString(response, logLine);
+            headersToString(response, logLine);
             log.debug(logLine.toString());
         } else {
             log.info(logLine.toString());
         }
     }
 
-    public void log(HttpServerRequest request, HttpServerResponse response, long requestStartTime) {
+    public void log(Logger logger, HttpServerRequest request, HttpServerResponse response, long requestStartTime) {
         logRequestResponse(request, response, requestStartTime, logger);
     }
+
+    public void addHeaderTransformationOutgoing(String header, UnaryOperator<String> transformation) {
```

Comment:
when configuring the transformation of headers in the application, are we supposed to use this method and the one for incoming?
If so, should we not document their purpose for devs of the framework?

Reply:
Yes, we really should. Odd that checkstyle didn't complain as it did on so much else. :P